### PR TITLE
tidb: rename products

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

- Remove the use of "on-premises" and similar ones
- Update "TiDB Cloud Serverless Tier" and similar ones to "TiDB Serverless"
- Update affected anchor links

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/14062
- Other reference link(s): https://github.com/pingcap/docs/pull/13668

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-6.0.0-dmr.md [hunk 1]
```diff
@@ -266,7 +266,7 @@ TiDB v6.0.0 is a DMR, and its version is 6.0.0-DMR.
 
 - An enterprise-level database management platform, TiDB Enterprise Manager
 
-    TiDB Enterprise Manager (TiEM) is an enterprise-level database management platform based on the TiDB database, which aims to help users manage TiDB clusters in on-premises or public cloud environments.
+    TiDB Enterprise Manager (TiEM) is an enterprise-level database management platform based on the TiDB database, which aims to help users manage TiDB clusters in  self-hosted or public cloud environments.
```
Comment (ran-huang) [inline releases/release-6.0.0-dmr.md:269]:
```suggestion
    TiDB Enterprise Manager (TiEM) is an enterprise-level database management platform based on the TiDB database, which aims to help users manage TiDB clusters in self-hosted or public cloud environments.
```

File: releases/release-7.0.0.md [hunk 2]
```diff
@@ -248,7 +248,7 @@ In v7.0.0-DMR, the key new features and improvements are as follows:
 * [DBeaver](https://dbeaver.io/) v23.0.1 supports TiDB by default [#17396](https://github.com/dbeaver/dbeaver/issues/17396) @[Icemap](https://github.com/Icemap)
 
     - Provides an independent TiDB module, icon, and logo.
-    - The default configuration supports [TiDB Cloud Serverless Tier](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier-beta), making it easier to connect to Serverless Tier.
+    - The default configuration supports [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier-beta), making it easier to connect to TiDB Serverless.
```
Comment (ran-huang) [inline releases/release-7.0.0.md:251]: Cloud 没有改 release notes，TiDB 的要改吗？

**Review by ran-huang — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@qiancai @ran-huang https://github.com/pingcap/docs/pull/13692/commits/17bbb93091bca2bf327b84d7f41bf54be2877371 PTAL

**Pull request merged**